neovim/api/buffer.py

```diff
@@ -183,3 +190,51 @@ def _normalize_index(self, index):
             if index > self.end:
                 index = self.end
         return index
+
+
+class Region(object):
+    def __init__(self, buffer, startmark, endmark):
+        self._range = buffer.range(startmark[0], endmark[0] + 1)
+        self.slice = slice(startmark[1], endmark[1] + 1)
```

Comment:
this is not a slice as `startmark[1]` and `endmark[1]` are indicies into different sequences (lines in this case).

Reply:
hm, this implementation is actually intended for visual block selection. This is indeed not sufficient for normal visual selection, when thinking about it.
How can a simple visual selection and a visual block selection be differentiated? Start- and end-marks are not sufficient for that, obviously, as it starts both of them.